firestore/src/include/firebase/firestore/load_bundle_task_progress.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef FIREBASE_FIRESTORE_CLIENT_CPP_SRC_INCLUDE_FIREBASE_FIRESTORE_LOAD_BUNDLE_TASK_PROGRESS_H_
+#define FIREBASE_FIRESTORE_CLIENT_CPP_SRC_INCLUDE_FIREBASE_FIRESTORE_LOAD_BUNDLE_TASK_PROGRESS_H_
+
+#include <cstdint>
+
+namespace firebase {
+namespace firestore {
+
+class LoadBundleTaskProgressInternal;
+
+/** Represents a progress update or a final state from loading bundles. */
+class LoadBundleTaskProgress {
+ public:
+  /**
+   * Represents the state of bundle loading tasks.
+   *
+   * Both `kSuccess` and `kError` are final states: task will abort
+   * or complete and there will be no more updates after they are reported.
+   */
+  enum class State { kError, kInProgress, kSuccess };
+
+  LoadBundleTaskProgress() = default;
+  LoadBundleTaskProgress(int32_t documents_loaded, int32_t total_documents,
+                         int64_t bytes_loaded, int64_t total_bytes,
+                         State state);
+
+  /** Returns how many documents have been loaded. */
+  int32_t documents_loaded() const { return documents_loaded_; }
+
+  /**
+   * Returns the total number of documents in the bundle. Returns 0 if the
+   * bundle failed to parse.
+   */
+  int32_t total_documents() const { return total_documents_; }
+
+  /** Returns how many bytes have been loaded. */
+  int64_t bytes_loaded() const { return bytes_loaded_; }
+
+  /** Returns the total number of bytes in the bundle. Returns 0 if the bundle
+   * failed to parse. */
+  int64_t total_bytes() const { return total_bytes_; }
+
+  /** Returns the current state of the loading progress. */
+  State state() const { return state_; }
+
+ private:
+  friend class EventListenerInternal;
+  friend class LoadBundleTaskProgressInternal;
+  friend struct ConverterImpl;
+
+  explicit LoadBundleTaskProgress(LoadBundleTaskProgressInternal* internal);
+
+  int32_t documents_loaded_;
+  int32_t total_documents_;
+  int64_t bytes_loaded_;
+  int64_t total_bytes_;
+  State state_;
+};
+
+inline bool operator==(const LoadBundleTaskProgress lhs,
```

Comment:
Is `lhs` supposed to be a reference? Also in `operator!=`.

Reply:
Thanks!